### ISSUE TITLE
Backendless discussion

### ISSUE THREAD

**Issue by relatedcode** [relatedcode/ParseAlternatives#31]

**Comment by vtongcoder** [comment]
Sorry for stupid question, I'm new on this backend service.
Anyone is using this?
Is it good and stable?
Thanks

**Comment by scadgek** [comment]
@vtongcoder Yup sure it is stable :) We've got a lot of users and even more now after the Parse's announcement.
We have a support desk, where you can share any difficulties you may run into when using our services and our team will try to respond you and help as fast as possible: http://support.backendless.com/
And also we've prepared a migration guide solely for former Parse users: https://backendless.com/parse-to-backendless-migration-guide/

**Comment by mauriciord** [comment]
I'm tried to create a column at Dashboard for a Data Table and didn't find any button to make that.

**Comment by markpiller** [comment]
@mauriciord here's how you do it:
Step 1: https://monosnap.com/file/2WL5RvTjdSZwKPtjqb8Klz59oiuCat
Step 2: https://monosnap.com/file/jYq27LyulcXJYL60sVeeVfqL1RVTj4

**Comment by ps-jimmy** [comment]
I saw the the 100 active users limit on the cloud functional limit page a few days ago but when i checked yesterday, it was off the list. Is the 100 active users limitation under review? Clarify this because its the only thing holding me back for now.

**Comment by markpiller** [comment]
We decided to remove the limitation on the number of active users.

**Comment by ps-jimmy** [comment]
Thanks Mark...Backendless is the best service so far...It has all that parse had and more. I am happy to come across it. Wish it stays forever.

**Comment by florinzf** [comment]
Hi. I was looking into Backendless a while ago and it did not have any documentation for Swift. Is this still the case? If so, are you guys planning on releasing one in the near future? What would be the best way to start using Backendless and Swift?

Thanks.

**Comment by ps-jimmy** [comment]
@florinzf Backendless has swift documentation now. The image below confirms that. @markpiller and his backendless team have done a good work on that.

![swift](https://cloud.githubusercontent.com/assets/5190294/12875400/643181d4-cde1-11e5-94f4-3e4815c094ec.png)

**Comment by markpiller** [comment]
@ps-jimmy and @florinzf this is work in progress, we will be updating the majority of the docs with Swift this week. The [getting started guide](backendless.com/mobile-developers/quick-start-guide-for-ios) already includes instructions for Swift. Also, there is a bunch of samples here: https://github.com/Backendless/BlogFeatureDay-iOS. These samples correspond to the articles in the [API Cookbook](https://backendless.com/blog-a-feature-a-day-challenge/)

**Comment by florinzf** [comment]
@markpiller Thanks a lot!

**Comment by cleyden** [comment]
I recall seeing Backendless was close to releasing a Unity plugin, what's the latest on that?

**Comment by markpiller** [comment]
@cleyden It is very high on our priority list. We're updating our iOS docs right now and will switch to finalizing Unity right after that.

**Comment by duracellrabbid** [comment]
Is there plans to release a plugin for Xamarin?

**Comment by foamx** [comment]
I'm starting to use backendless and look forward to you guys getting the SWIFT docs and and examples completed.  If it were HIPAA compliant that would be the icing on the cake!  Any chance of getting a BAA in place for Backendless?

**Comment by markpiller** [comment]
@duracellrabbid Yes, it is on the roadmap. 

@foamx The Managed Backendless product is HIPAA compliant. You can also use Standalone Backendless and get it certified yourself. What do you mean by "getting a BAA in place for Backendless"?

**Comment by foamx** [comment]
@markpiller http://searchhealthit.techtarget.com/definition/HIPAA-business-associate-agreement-BAA

**Comment by foamx** [comment]
@markpiller Great news that the Managed Backend is HIPAA compliant!

**Comment by duracellrabbid** [comment]
@markpiller Thanks for the reply. Is there an E.T.C. for the plugin? I am considering using Backendless for my latest project on Xamarin. While I know I can use the REST API, having a plugin would save me a lot of work.

**Comment by markpiller** [comment]
@duracellrabbid There is a user who is working on creating Xamarin SDK based on our SDK for .NET. Here's his latest post: http://support.backendless.com/topic/xamarin-sdk

**Comment by meoz** [comment]
@markpiller I thinking about using Backendless for a mobile/web app. Did not found if the iOS SDK supports offline syncing, do you have any info about this ?

**Comment by ahpiau18** [comment]
i am migrating all exisitng apps to backendless, and i found it is great. Overall speed (data returned from API) is faster than Parse.
One thing i found it is missing is, the sql filter is not as user friendly as parse.
After filtered the data, and i clicked on column header for sorting, it reload all the data again.
Hope this can be improve soon.
:)

**Comment by markpiller** [comment]
@meoz offline syncing is coming this summer.
@ahpiau18 thanks, we will look into it.

**Comment by Granfalloner** [comment]
Recently, I decided to transfer to Backendless a simple app with quite limited backend need. So, the first impression was - its really powerful thing! **BUT...**

1. Way too much bugs. I personally reported a few in just a couple of days (including crash) - and thats only for one platform!
2. Whats even worse, some of the bugs I encountered had broke already working code of other people (according to comments). So you can never be sure.
3. Documentation for SDK is far from full, a lot of methods are not covered. And some important behaviour is not documented at all - you have to search through support database to find why this thing is not working the way you would expect.
4. Support likes to argue with you and make critical comments instead of doing there job. I was very surprised with such attitude - I used to much more polite responses than that.

So, to summarise, the initial impression was good, but after more thorough examining I have encountered too many problems. Service is not stable and reliable. Also, its not very intuitive in use (what becomes even worse because of a bad documentation). Unfortunately, for now, I cannot recommend it for using. But nevertheless, I hope it will change to better in the future - it has quite a big potential.

**Comment by markpiller** [comment]
@Granfalloner it is pretty obvious that your goal is to trash us. I am not sure what the reason is, I suspect you were not happy that I asked you to be more polite on the forum - you were quite emotional, but that's ok. More than half of your statements are either not true or exaggerated. As I said, you can have an opinion, but plenty of other people would strongly disagree with you on every single point: documentation, stability, intuitiveness of the APIs.

**Comment by Granfalloner** [comment]
@markpiller, "trash you"? I spent (not to say wasted) a few days rewriting an app for your service, found a lot of issues, felt very unsatisfied, did a few reports (thus serving you as a tester - you should appreciate that) but surely, "it is pretty obvious" that the sole purpose of my actions was just to "thrash you". But in fact, after I spend more time fixing or trying to workaround some issues in your SDK than I spend on the app itself, my review is actually quite mild. And yes, plenty of people would disagree with me, but plenty of people already did found that I am actually right in my statements - and you can easily recall such examples from recently logged problems on support page.
But, still, I don't say your service is trash: the potential is great, and the first impression is very good. But, its not as user-friendly and stable as one would expect. Thus you can use it at your own risk, but for those concerned with reliability I wouldn't advise to do so now. Probably, service will change to the better in the future. But only in case if support starts to solve reported problem and improve the service, and not trying to explain negative feedbacks like "you were not happy that I asked you to be more polite".

**Comment by scadgek** [comment]
@Granfalloner You should also understand, that Backendless is not sponsored by some technological giant like Firebase by Google or Parse by Facebook. This at the same time is our advantage, as we don't depend on someone else's decisions, but also a drawback - as we don't have such a huge funding and working resources support.

Considering this, we are a pretty small team working on a rather big project with a countless number of features (I'm sure you would agree with that). So we have to put priorities properly and stay focused on the most required features and fixing the most critical bugs, though, besides that, we also try to be as responsive and reactive to customers' requests as possible. That's why sometimes we don't really have much time to investigate your problem right at the moment, but be sure that it is assigned to one of our developers and you may receive information about the progress in the same support topic if you ask.

And if you're also a developer, I bet you understand that there are always bugs in any software. And the bigger it is, the more you can find. As you might have seen, Backendless is a huge one, but we're keeping the issues prioritized and fix them as soon as possible, while also constantly providing our users and customers some new useful features.

**Comment by Granfalloner** [comment]
@markpiller Good to see polite response with explanations instead of blaming someone.
I understand the complexity of task, and I hope you are moving in a right direction. But, as developer, I do value reliable things. That is, I cannot afford myself to fix bugs that I didn't make. Unfortunately, this is not the case with Backendless now.

**Comment by anhourglass** [comment]
Backendless has been working great so far. Something that stood out to me was the helpfulness of the devs. I see them replying to a ton of posts in their support forum constantly.  I had an issue once that was completely me just being stupid and Mark replied super quickly with a helpful response. That left an impression on me that they really care about their users.

**Comment by Scitator** [comment]
@markpiller what about kotlin support?
Yes, java for android, but you know, koltin seems to be better choise.

**Comment by Scitator** [comment]
@scadgek You know, maybe it's a good idea, I know nothing, what to do, but it can be great experience

Yes, I can, but Kotlin code look more friendly, than kotlin-java one

**Comment by JamesHongBiz333** [comment]
hello, guys,

Anyone can introduce or give some idea about how or if it's possible to migration from Parse to Backendless ?

**Comment by scadgek** [comment]
@JamesHongBiz333 We have prepared a Parse Migration Guide earlier, hope it helps: https://backendless.com/parse-to-backendless-migration-guide/
